src/Giraffe/HttpContextExtensions.fs

```diff
@@ -61,13 +61,21 @@ type HttpContextExtensions() =
         loggerFactory.CreateLogger categoryName
 
     /// <summary>
-    /// Gets an instance of <see cref="Microsoft.Extensions.Hosting.IHostingEnvironment"/> from the request's service container.
+    /// This function is deprecated and it's going to be removed in future releases! Gets an instance of <see cref="Microsoft.Extensions.Hosting.IHostingEnvironment"/> from the request's service container.
     /// </summary>
     /// <returns>Returns an instance of <see cref="Microsoft.Extensions.Hosting.IHostingEnvironment"/>.</returns>
     [<Extension>]
     static member GetHostingEnvironment(ctx : HttpContext) =
```

Comment:
Could you add the [Obsolete](https://learn.microsoft.com/en-us/dotnet/api/system.obsoleteattribute?view=net-7.0) attribute to this?

Reply:
Sure, thanks for the heads up!